src/AzureAuth/CommandMain.cs

```diff
@@ -262,10 +262,45 @@ private int GetToken()
             try
             {
                 ITokenFetcher tokenFetcher = this.TokenFetcher();
-                TokenResult tokenResult = (this.tokenFetcherOptions.Scopes == null
-                    ? tokenFetcher.GetAccessTokenAsync(this.CombinedAuthMode)
-                    : tokenFetcher.GetAccessTokenAsync(this.tokenFetcherOptions.Scopes, this.CombinedAuthMode))
-                    .Result;
+                TokenResult tokenResult = null;
+
+                // When running multiple AzureAuth processes with the same tenant and resource ID,
+                // They may prompt many times, which is annoying and unexpected.
+                // Use Mutex to ensure that only one process can access the corresponding resource at the same time.
+                string lockName = "Global\\"
+                    + this.Resource
+                    + this.Client
+                    + this.Tenant;

```

Comment:
This is just my opinion, but it might be a little cleaner to use string interpolation here. Also, consider using an underscore to separate each of these IDs in case the lock name needs to be read for debugging later.
```suggestion
                string lockName = $"Global\\{this.Resource}_{this.Client}_{this.Tenant}";
```

Reply:
Yet another consideration would be to hash (maybe with a SHA algorithm?) all three values so that they aren't accidentally leaked anywhere else in the OS (I don't actually know what C# does with the lock name). I'm not sure if that's necessary as I don't know if this trio is really considered "secret", but it might be worth thinking about.